packages/admin-ui/src/filters/filter-multiple.state.tsx

```diff
@@ -1,11 +1,82 @@
-import type {
-  UseFilterMultipleReturn,
-  UseFilterMultipleStateProps,
-} from './base-filter.state'
-import { useBaseFilterState } from './base-filter.state'
-
-export function useFilterMultipleState(
-  props: UseFilterMultipleStateProps
-): UseFilterMultipleReturn {
-  return useBaseFilterState({ ...props, selectionMode: 'multiple' })
+import { useState, useCallback, useEffect } from 'react'
+
+import type { ComboboxMultipleState } from '../combobox'
+import { useComboboxMultipleState } from '../combobox'
+import { useMenuState } from 'ariakit/menu'
+import type { GenericFilterStateReturn, FilterItem } from './filter.state'
+
+export function useFilterMultipleState<T extends FilterItem>(
+  props: UseFilterMultipleStateProps<T>
+): UseFilterMultipleReturn<T> {
+  const { items, label, initialApplied, onChange = () => {} } = props
+
+  const [appliedItems, setAppliedItems] = useState<T[]>([])
+
+  const comboboxMultiple = useComboboxMultipleState<T>({
+    getOptionValue: (op) => op.id,
+  })
+
+  useEffect(() => {
+    const initialItem = items.filter((it) => initialApplied?.includes(it.id))
+
+    comboboxMultiple.setSelectedItems(initialItem)
+    setAppliedItems(initialItem)
+  }, [])
+
+  const menu = useMenuState(comboboxMultiple)
+
+  const { selectedItems } = comboboxMultiple
+
+  const apply = useCallback(() => {
+    setAppliedItems(selectedItems)
+
+    onChange({ selected: selectedItems.map((i) => i.id) })
+    menu.hide()
+  }, [onChange])
+
+  const clear = useCallback(() => {
+    comboboxMultiple.clearSelected()
+
+    setAppliedItems([])
+    onChange({ selected: [] })
+    menu.hide()
+  }, [onChange])
+
+  useEffect(() => {
+    // Resets combobox value when menu is closed
```

Comment:
Maybe you could let the following expression more explicit and maybe avoid the need of the comment line.

Expression: `!menu.mounted && (comboboxMultiple.value || selectedItems?.length)`

Suggestion:
```suggestion
   const isMenuClosed = !menu.mounted
   const hasItemSelected = (comboboxMultiple.value || selectedItems?.length)
   
   if(isMenuClosed && hasItemSelected ) {
       ...
   }
```